app/src/androidTest/java/com/github/se/eventradar/viewProfile/ProfileTest.kt

```diff
@@ -30,22 +35,40 @@ class ProfileTest : TestCase(kaspressoBuilder = Kaspresso.Builder.withComposeSup
   // Relaxed mocks methods have a default implementation returning values
   @RelaxedMockK lateinit var mockNavActions: NavigationActions
 
-  @RelaxedMockK lateinit var mockProfileViewModel: ProfileViewModel
-
-  private val sampleUiState =
-      MutableStateFlow(
-          ProfileUiState(
-              profilePicUrl = "",
-              firstName = "Jim",
-              username = "Smith",
-              bio = "I am Jim Smith and I love the Smiths (the band)."))
+  private lateinit var mockUserRepository: MockUserRepository
+  private lateinit var mockProfileViewModel: ProfileViewModel
 
   @Before
-  fun testSetup() {
-    every { mockProfileViewModel.getProfileDetails() } returns Unit
-    every { mockProfileViewModel.uiState } returns sampleUiState
+  fun testSetup() = runTest {
+    mockUserRepository = MockUserRepository()
+
+    mockUserRepository.updateCurrentUserId("1")
+
+    mockUserRepository.addUser(
+        User(
+            userId = "Test",
+            birthDate = "01/01/2000",
+            email = "",
+            firstName = "Test",
+            lastName = "Test",
+            phoneNumber = "123456789",
+            accountStatus = "active",
+            eventsAttendeeList = mutableListOf(),
+            eventsHostList = mutableListOf(),
+            friendsList = mutableListOf(),
+            profilePicUrl =
+                "https://firebasestorage.googleapis.com/v0/b/event-radar-e6a76.appspot.com/o/Profile_Pictures%2Fplaceholder.png?alt=media&token=ba4b4efb-ff45-4617-b60f-3789e8fb75b6",
+            qrCodeUrl = "",
+            bio = "",
+            username = "1"))
+
+    mockProfileViewModel = ProfileViewModel(mockUserRepository, "1")
+
+    mockProfileViewModel.getProfileDetails()
   }
 
+  @After fun testTeardown() = runTest { unmockkAll() }
+
   @Test
   fun screenDisplaysAllElementsCorrectlyWhenPublic() = run {
```

Comment:
Nit: The name of this is a little confusing, what do you mean by "WhenPublic"?

Reply:
As the description explains, I have merged both the Profile page and the Friends' View Profile, to limit code redundancy. In the test and in the code this is portrayed by the isPublicView boolean that indicates if another user is seeing my "Public" profile or if I am seeing my "Private" Profile

Reply:
Gotcha, thanks for the explanation.